ssl/t1_lib.c

```diff
@@ -1692,7 +1693,8 @@ unsigned char *ssl_add_serverhello_tlsext(SSL *s, unsigned char *buf,
         s2n(0, ret);
     }
 
-    if (s->s3->alpn_selected) {
+    /* send only if proposed to us and selected */
```

Comment:
You're right, it should never happen; the only way it could happen is if the application were mucking with these values, which can no longer happen in 1.1 with opaque data structures.